### PR TITLE
Update Python SDK to match Node SDK

### PR DESCRIPTION
**Pull request by ameyakhare**
Resolves AIR-3359

## Description
Adds feature parity between python sdk and node sdk. Deprecates run in favor of execute.

{'id': 'run20220603zhlxlfncy4c', 'taskID': 'tsk20220418z8re5w1ea71', 'paramValues': {}, 'status': 'Succeeded', 'output': [{'environment': 'prod', 'message': 'Testing in prod!'}]}

### CODE REVIEW

File: airplane/__init__.py [hunk 1]
```diff
@@ -40,6 +40,11 @@ def write_named_output(name: str, value: Any) -> None:
     return __deprecated_write_named_output(name, value)
 
 
+@deprecation.deprecated(
+    deprecated_in="0.3.2",
+    current_version=__version__,
+    details="Use execute(task_id, paramValues) instead.",
```
Comment (colinking) [inline airplane/__init__.py:46]:
```suggestion
    details="Use execute(task_slug, paramValues) instead.",
```

File: airplane/__init__.py [hunk 2]
```diff
@@ -49,26 +54,41 @@ def run(
...
 
 @backoff.on_exception(
-    backoff.expo(factor=0.1, max_value=5),
+    lambda: backoff.expo(factor=0.1, max_value=5),
     (
         requests.exceptions.ConnectionError,
         requests.exceptions.Timeout,
         RunPendingException,
     ),
     max_tries=1000,
```
Comment (colinking) [inline airplane/__init__.py:84]: To match the semantics of the Node SDK, set max tries to infinite (the run will timeout at some point anyway).

File: airplane/client.py [hunk 3]
```diff
@@ -38,15 +38,28 @@ def create_run(
...
+                "slug": slug,
+                "paramValues": param_values,
+            },
+            headers=self._headers,
+        )
+        self.__maybe_error_on_response(resp)
+        return resp.json()["runID"]
+
+    def get_run(self, run_id: str) -> Dict[str, Any]:
         """Gets the status of an Airplane task run."""
```
Comment (colinking) [inline airplane/client.py:55]:
```suggestion
        """Fetches an Airplane task run."""
```

File: airplane/__init__.py [hunk 4]
```diff
@@ -49,26 +54,41 @@ def run(
...
+    outputs = client.get_run_outputs(run_id)
+    return {"status": run_info["status"], "outputs": outputs}
+
+
+def execute(slug: str, param_values: Optional[Dict[str, Any]] = None) -> Dict[str, Any]:
+    """Triggers an Airplane task with the provided arguments."""
+    client = api_client_from_env()
+    run_id = client.execute_task(slug, param_values)
+    run_info = __wait_for_run_completion(run_id)
     outputs = client.get_run_outputs(run_id)
```
Comment (colinking) [inline airplane/__init__.py:67]:
```suggestion
    output = client.get_run_output(run_id)
```

**Review by colinking — APPROVED** [review]
lgtm

File: airplane/client.py [hunk 5]
```diff
@@ -38,15 +38,28 @@ def create_run(
...
         resp = requests.get(
             f"{self._api_host}/v0/runs/get",
             params={"id": run_id},
             headers=self._headers,
         )
         self.__maybe_error_on_response(resp)
```
[the remark this answers is not in the document]
Comment (ameyakhare) [reply airplane/client.py:61]: we use the error field in the response, so would rather leave this for now

**Pull request merged**